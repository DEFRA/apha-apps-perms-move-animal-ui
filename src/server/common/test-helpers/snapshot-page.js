import { parseDocument } from '../../common/test-helpers/dom.js'
import { createServer } from '~/src/server/index.js'
import { withCsrfProtection } from './csrf.js'
/** @import { Server } from '@hapi/hapi' */

/* global expect, it, beforeAll, afterAll, describe */

/**
 * @param {string} describeMessage
 * @param {string} shouldMessage
 * @param {string} urlPath
 */
export const describePageSnapshot = (
  describeMessage,
  shouldMessage,
  urlPath
) => {
  describe(describeMessage, () => {
    /** @type {Server} */
    let server

    beforeAll(async () => {
      server = await createServer()
      await server.initialize()
    })

    afterAll(async () => {
      await server.stop({ timeout: 0 })
    })

    it(shouldMessage, async () => {
      const { payload } = await server.inject(
        withCsrfProtection({
          method: 'GET',
          url: urlPath
        })
      )

      const content =
        parseDocument(payload).querySelector('#main-content')?.innerHTML

      expect(content).toMatchSnapshot()
    })
  })
}
