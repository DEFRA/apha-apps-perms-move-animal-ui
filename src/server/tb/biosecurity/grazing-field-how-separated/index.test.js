import { describePageSnapshot } from '../../../common/test-helpers/snapshot-page.js'
import { manureAndSlurryDetailsPage } from '../manure-and-slurry-details/index.js'
import {
  Answer,
  grazingFieldHowSeparatedPage,
  GrazingFieldHowSeparatedPage
} from './index.js'

const sectionKey = 'biosecurity'
const question =
  'Which measures are being taken to reduce the spread of TB when the animals are grazing?'
const questionKey = 'grazingFieldHowSeparated'
const pageUrl = '/biosecurity/grazing-field-how-separated'

describe('GrazingFieldHowSeparatedPage', () => {
  const page = new GrazingFieldHowSeparatedPage()

  it('should have the correct urlPath', () => {
    expect(page.urlPath).toBe(pageUrl)
  })

  it('should have the correct sectionKey', () => {
    expect(page.sectionKey).toBe(sectionKey)
  })

  it('should have the correct question', () => {
    expect(page.question).toBe(question)
  })

  it('should have the correct questionKey', () => {
    expect(page.questionKey).toBe(questionKey)
  })

  it('should have the correct Answer model', () => {
    expect(page.Answer).toBe(Answer)
  })

  it('nextPage should return any shared buildings page', () => {
    const nextPage = page.nextPage(
      new Answer({
        grazingFieldHowSeparated: ['three']
      })
    )
    expect(nextPage).toBe(manureAndSlurryDetailsPage)
  })

  it('should export page', () => {
    expect(grazingFieldHowSeparatedPage).toBeInstanceOf(
      GrazingFieldHowSeparatedPage
    )
  })

  describePageSnapshot({
    describes: 'grazingFieldHowSeparatedPage.content',
    it: 'should render expected response and content',
    pageUrl
  })
})
