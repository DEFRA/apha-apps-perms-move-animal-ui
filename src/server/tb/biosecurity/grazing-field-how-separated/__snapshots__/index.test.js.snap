// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GrazingFieldHowSeparatedPage grazingFieldHowSeparatedPage.content should render expected response and content 1`] = `
"
  <h1 class="govuk-heading-l">Separating the incoming animals</h1>
  <div class="govuk-grid-row">
    <div class="govuk-grid-column-two-thirds">
      <p>
        We need to know what measures are in place to reduce the risk of
        spreading TB to the incoming animals from the resident herd.
      </p>
      <p>You should include:</p>
      <ul class="govuk-list govuk-list--bullet">
        <li>
          minimum distances between incoming animals and the resident herd
        </li>
        <li>the types of fencing you use</li>
        <li>how grazing fields are kept separate</li>
      </ul>
    </div>
  </div>
    <form method="POST" novalidate="">
      <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds">
          <input type="hidden" name="crumb" value="csrf-value">
          <input type="hidden" name="nextPage" value="">

<fieldset class="govuk-fieldset">
  <div class="govuk-form-group">
  <label class="govuk-label govuk-label--m" for="grazingFieldHowSeparated">
    How will you separate the incoming animals from the resident herd?
  </label>
  <textarea class="govuk-textarea" id="grazingFieldHowSeparated" name="grazingFieldHowSeparated" rows="8"></textarea>
</div>


</fieldset>
            <button type="submit" class="govuk-button" data-module="govuk-button" id="continue-button">
  Continue
</button>
        </div>
      </div>
    </form>
        "
`;
